text/0074-alphanumeric-transfer-ids.md

```diff
@@ -0,0 +1,130 @@
+# Alphanumeric Transfer Id's
+
+## Summary
+
+[summary]: #summary
+
+CEP PR: [casperlabs/ceps#0074](https://github.com/casperlabs/ceps/pull/0074)
+
+In summary, transfer ID's will be extended to accept alphanumeric strings. Currently, transfer IDs are an optional integral type. The goal of this CEP is to update transfer IDs to be a string of alphanumeric characters, an integer or empty.
+
+## Motivation
+
+[motivation]: #motivation
+
+The main motivation for this CEP is to make integration with the Casper Blockchain Network easier for developers and organizations. As an example, if a development or accounting team uses an alphanumeric ID system for their in-house record keeping, they would be able to continue to use the same system, which could make it easier to reference on-chain transactions in their data.
+
+## Guide-level explanation
+
+[guide-level-explanation]: #guide-level-explanation
+
+This CEP introduces a new type that will extend the `id` field of the `Transfer` struct to allow it to store alphanumeric string-based memos: `Memo`.
+
+Alphanumeric strings will contain the characters `0` through `9`, and `a` through `z`. Alphabetic characters can be uppercase or lowercase. Transfer IDs will no longer only be optional integral types, so code that reads or writes transfer IDs will need to be updated.
+
+## Reference-level explanation
+
+[reference-level-explanation]: #reference-level-explanation
+
+Here is an example implementation for a new `Memo` type that will be stored in a `Transfer`'s `id` field.
+
+```rust
+/// Memo field associated with a [`Transfer`].
+#[derive(Debug, Clone, Ord, PartialOrd, Eq, PartialEq, Serialize, Deserialize)]
+#[cfg_attr(feature = "json-schema", derive(JsonSchema))]
+pub enum Memo {
+    None,
+    Numeric(u64),
+    Alphanumeric(String),
```

Comment:
Will there be a limit on the size of this? Or is it going to be unbounded? What encoding will it use, utf-8?